src/modules/Cart/Service.php

```diff
@@ -47,31 +47,84 @@ public function getSearchQuery($data)
      * @return \Model_Cart
      */
     public function getSessionCart()
-    {
-        $sqlBindings = [':session_id' => $this->di['session']->getId()];
-        $cart = $this->di['db']->findOne('Cart', 'session_id = :session_id', $sqlBindings);
-
-        if ($cart instanceof \Model_Cart) {
-            return $cart;
-        }
-
-        $cc = $this->di['mod_service']('currency');
-
-        if ($this->di['session']->get('client_id')) {
-            $client_id = $this->di['session']->get('client_id');
-            $currency = $cc->getCurrencyByClientId($client_id);
+    {   
+        $secret_key = $this->di['db']->getCell('SELECT value FROM setting WHERE param = :param', [':param' => 'ssh_key']);
+        $secret_iv = $this->di['db']->getCell('SELECT value FROM setting WHERE param = :param', [':param' => 'ssh_iv']); 
+        $session_id = $this->di['session']->getId();
+        $client_id = $this->di['session']->get('client_id');
+    
+        $guestCartToken = isset($_COOKIE['gcid']) ? $this->decrypt($_COOKIE['gcid']) : null;
+    
+        $clientCart = null;
+        if ($client_id) {
+            $clientCart = $this->di['db']->findOne('Cart', 'client_id = :client_id', [':client_id' => $client_id]);
+        }
+    
+        $sessionCart = null;
+        if ($guestCartToken) {
+            $sessionCart = $this->di['db']->findOne('Cart', 'session_id = :session_id AND client_id IS NULL', [':session_id' => $guestCartToken]);
+        }
+    
+    if ($clientCart && $sessionCart) {
+        $this->mergeCarts($clientCart, $sessionCart);
+        return $clientCart;
+    }
+        if ($clientCart) {
+            return $clientCart;
+        }
+        if ($sessionCart) {
+            if ($client_id) {
+                $sessionCart->client_id = $client_id;
+                $this->di['db']->store($sessionCart);
+            }
+            return $sessionCart;
+        }
+        $currency = $this->di['mod_service']('currency')->getDefault();
+        $newCart = $this->di['db']->dispense('Cart');
+        $newCart->session_id = $session_id;
+        $newCart->client_id = $client_id ?: null;
+        if (isset($profile) && isset($profile->currency) && isset($profile->currency->id)) {
```

Comment:
This $profile variable is never defined and I am *fairly* certain that `$profile->currency->id` will never work